server/sql.go

```diff
@@ -992,3 +1011,44 @@ func isMySQLExcludeDatabase(database string) bool {
 	}
 	return true
 }
+
+func (s *Server) allowAccessDatabase(ctx context.Context, principalID int, instanceID int, databaseName string) (bool, error) {
+	database, err := s.getDatabase(ctx, instanceID, databaseName)
+	if err != nil {
+		if httpError, ok := err.(*echo.HTTPError); ok && httpError.Code == echo.ErrNotFound.Code {
+			return true, nil
+		}
+		return false, err
+	}
+
+	isDeveloper := false
+	for _, member := range database.Project.ProjectMemberList {
```

Comment:
Any possibility that the principal doesn’t exist in the `ProjectMemberList`, then finally, we will treat it as `isDeveloper == false`?